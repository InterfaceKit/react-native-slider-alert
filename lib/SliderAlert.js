/* @flow */

import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Animated, Text } from 'react-native';

type Props = {
  startValue: number,
  value: number,
  containerStyle: number | Object,
  titleStyle: number | Object,
  delay: number,
  duration: number,
  children?: string | React.ReactNode | React.ReactElement<any>
};
type State = {
  message: string
};

// WIP
class SliderAlert extends React.PureComponent<Props, State> {
  _height = new Animated.Value(0);

  static defaultProps = {
    delay: 2250,
    duration: 250
  };

  static propTypes = {
    startValue: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
    containerStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    titleStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    delay: PropTypes.number.isRequired,
    duration: PropTypes.number.isRequired,
    children: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.node,
      PropTypes.element
    ])
  };

  state = {
    message: ''
  };

  showAlert = (message: string, onAnimationEnd?: Function) => {
    const { value, duration, startValue, delay } = this.props;
    if (typeof message !== 'string') {
      console.warn(
        'Message is required on SliderAlert or should be a string type!'
      );
      return;
    }
    this.setState({ message }, () => {
      Animated.sequence([
        Animated.timing(this._height, {
          toValue: startValue,
          duration: duration
        }),
        Animated.timing(this._height, {
          toValue: value,
          delay: delay
        })
      ]).start(() => {
        if (typeof onAnimationEnd === 'function') {
          onAnimationEnd();
        }
      });
    });
  };

  render() {
    return (
      <Animated.View
        style={[
          styles.container,
          this.props.containerStyle,
          {
            height: this._height
          }
        ]}>
        {this.props.children || (
          <Text style={[styles.titleStyle, this.props.titleStyle]}>
            {this.state.message}
          </Text>
        )}
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F5FCFF',
    alignItems: 'center',
    justifyContent: 'center'
  },
  title: {
    fontSize: 14
  }
});

export default SliderAlert;
