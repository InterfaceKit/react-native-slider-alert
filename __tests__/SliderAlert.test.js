import React from 'react';
import SliderAlert from '../lib/SliderAlert';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <SliderAlert
        startValue={40}
        value={0}
        containerStyle={{
          backgroundColor: '#F4D35E'
        }}
        titleStyle={{
          fontSize: 18,
          fontFamily: 'Avenir Next'
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
