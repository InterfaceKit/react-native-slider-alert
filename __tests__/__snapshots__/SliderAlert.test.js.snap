// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4D35E",
      "height": 0,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        undefined,
        Object {
          "fontFamily": "Avenir Next",
          "fontSize": 18,
        },
      ]
    }
  >
    
  </Text>
</View>
`;
